### PR TITLE
👉 Implement a `<MultiSelect />` component

### PR DESCRIPTION
**Pull request by thesan**
See #692 (the PR for this issue started to get too long)

### CODE REVIEW

**Comment by jodator** [comment]
@thesan Hey, could you check the errors? You can also see the latest changes on master as there were a few updates.

**Comment by thesan** [comment]
@jodator yes I fixed it already, but I realized some of the changes affect `<SelectAccount />` so I'm fixing this now.

File: packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ export const SelectAccount = React.memo(({ onChange, filter, selected }: Props)
   const filteredOptions = useMemo(() => filterByText(options, search), [search, options])
   const keyring = useKeyring()
 
+  useEffect(() => setSelectedOption(selected), [selected])
```
Comment (thesan) [inline packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx:36]: The changes broke the Account selection, but I don't understand how it use to work without this line. Am I missing something ?

Comment (jodator) [reply packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx:36]: Previously the idea was to allow passing a pre-defined account to the component. It is used in line 30 as a default selected option in `useState()`.

Comment (jodator) [reply packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx:36]: So there are two options
- passing account as `selected` -> the component is rendered with an already selected item
- passing `undefined` as `selected` -> the component is rendered as empty

Comment (thesan) [reply packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx:36]: I get this, but I'm confused because it seems that after the component has been mounted: `setSelectedOption()` was only called when `filteredOptions.length === 0`. So I don't understand how the value of `selectedOption` updated, unless the component got unmounted

Comment (jodator) [reply packages/ui/src/accounts/components/SelectAccount/SelectAccount.tsx:36]: The hook below (line 37) should be rewritten AFAICS, as it is hard to parse. _I think_ that it is used to set "unknown" account when user pastes an address to the search box which is not a known account.
